pkg/sources/gitlab/gitlab.go

```diff
@@ -293,45 +293,62 @@ func (s *Source) basicAuthSuccessful(apiClient *gitlab.Client) bool {
 func (s *Source) getAllProjects(ctx context.Context, apiClient *gitlab.Client) ([]*gitlab.Project, error) {
 	// Projects without repo will get user projects, groups projects, and subgroup projects.
 	user, _, err := apiClient.Users.CurrentUser()
-
 	if err != nil {
-		msg := fmt.Sprintf("unable to authenticate using: %s", s.authMethod)
-		return nil, errors.WrapPrefix(err, msg, 0)
+		return nil, fmt.Errorf("unable to authenticate using %s, %w", s.authMethod, err)
+	}
+
+	uniqueProjects := make(map[int]*gitlab.Project)
+	var (
+		projects              []*gitlab.Project
+		projectsWithNamespace []string
+	)
+
+	// Used to filter out duplicate projects.
```

Comment:
do we see these? (i guess we could if the project list is updated during the operation)

Reply:
Yea, i was initially thinking the same thing, but i think the same project can appear under the user's list as well as the group's list.